### ISSUE TITLE
Problem in multimodelcollator call function

### ISSUE THREAD

**Issue by one2clouds** [nothineazi/Visual-Question-Answering#1]
My MultimodalCollator class and MultimodalVQAModel class are same from yours.

In below code from yours : when i try to use trainer.train() is then it says, i['question'] is not defined. And when I print raw_batch_dict it just gives the list of dictionaries containing only labels. 
```
def __call__(self, raw_batch_dict):
        return {
            **self.tokenize_text(
                raw_batch_dict['question']
                if isinstance(raw_batch_dict, dict) else
                [i['question'] for i in raw_batch_dict]
            ),
            **self.preprocess_images(
                raw_batch_dict['image_id']
                if isinstance(raw_batch_dict, dict) else
                [i['image_id'] for i in raw_batch_dict]
            ),
            'labels': torch.tensor(
                raw_batch_dict['label']
                if isinstance(raw_batch_dict, dict) else
                [i['label'] for i in raw_batch_dict],
                dtype=torch.int64
            ),
        }
```
my createMultimodelVQACollatorAndModel function:
```
def createMultimodalVQACollatorAndModel(text='bert-base-uncased', image='google/vit-base-patch16-224-in21k'):
    tokenizer = AutoTokenizer.from_pretrained(text)
    preprocessor = AutoFeatureExtractor.from_pretrained(image)

    collator = MultimodalCollator(
        tokenizer=tokenizer,
        preprocessor=preprocessor,
    )
    model = MultimodalVQAModel(pretrained_text_name=text, pretrained_image_name=image).to(device)
    return collator, model
```
then defining collator and model
``` collator, model = createMultimodalVQACollatorAndModel(text='bert-base-uncased', image='google/vit-base-patch16-224-in21k') ```

my compute metrics is:
```
def compute_metrics(pred_label_tuple):
    logits,labels = pred_label_tuple
    preds = logits.argmax(axis=-1)
    return {
        "Accuracy": accuracy_score(labels,preds),
        "Precision": precision_score(labels,preds,average='macro'),
        "Recall": recall_score(labels,preds,average='macro'),
        "F1": f1_score(labels,preds,average='macro')
    }
```
training args:
```
args = TrainingArguments(
    output_dir = 'Shir-VQA-fuse',
    evaluation_strategy="epoch",
    save_strategy="epoch",
    per_device_train_batch_size=32,
    per_device_eval_batch_size=32,
    learning_rate=2e-5,
    num_train_epochs=3,
    weight_decay=0.01,
    push_to_hub=True,
    dataloader_num_workers=8,
    load_best_model_at_end=True
)
```
trainer:
```
trainer = Trainer(
    model,
    args,
    train_dataset= dataset["train"],
    eval_dataset= dataset["val"],
    data_collator=collator,
    compute_metrics=compute_metrics,
)
```
When i try to run this i get error:
```
train_multi_metrics = trainer.train() 
```

When I try to print raw_batch_dict then I just get this:
```
[{'labels': 1}, {'labels': 1}, {'labels': 873}, {'labels': 997}, {'labels': 640}, {'labels': 997}, {'labels': 997}, {'labels': 997}, {'labels': 574}, {'labels': 574}, {'labels': 995}, {'labels': 969}, {'labels': 369}, {'labels': 551}, {'labels': 816}, {'labels': 0}, {'labels': 997}, {'labels': 487}, {'labels': 997}, {'labels': 13}, {'labels': 22}, {'labels': 997}, {'labels': 997}, {'labels': 20}, {'labels': 640}, {'labels': 738}, {'labels': 574}, {'labels': 810}, {'labels': 997}, {'labels': 1}, {'labels': 848}, {'labels': 997}, {'labels': 997}, {'labels': 13}, {'labels': 391}, {'labels': 574}, {'labels': 997}, {'labels': 640}, {'labels': 738}, {'labels': 997}, {'labels': 330}, {'labels': 536}, {'labels': 997}, {'labels': 675}, {'labels': 13}, {'labels': 640}, {'labels': 145}, {'labels': 613}, {'labels': 574}, {'labels': 230}, {'labels': 997}, {'labels': 46}, {'labels': 574}, {'labels': 640}, {'labels': 583}, {'labels': 865}, {'labels': 997}, {'labels': 667}, {'labels': 49}, {'labels': 991}, {'labels': 281}, {'labels': 997}, {'labels': 997}, {'labels': 1}]
```
